exporter/kafkaexporter/pdata_marshaler.go

```diff
@@ -83,16 +86,25 @@ type pdataTracesMarshaler struct {
 }
 
 func (p pdataTracesMarshaler) Marshal(td pdata.Traces, topic string) ([]*sarama.ProducerMessage, error) {
-	bts, err := p.marshaler.MarshalTraces(td)
-	if err != nil {
-		return nil, err
-	}
-	return []*sarama.ProducerMessage{
-		{
+	var messages []*sarama.ProducerMessage
+	var errs error
+	for _, batch := range batchpersignal.SplitTraces(td) {
+		bts, err := p.marshaler.MarshalTraces(td)
+		if err != nil {
+			// continue to process batches that can be serialized
+			errs = multierr.Append(errs, err)
+			continue
+		}
+
+		// every batch should have at least one span
```

Comment:
I think it's safer to check the data to make sure than panic in case it happens to not be the case, e.g. I'm not sure if the OTLP receiver would filter out a ResourceSpans with no spans as required here